src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -297,6 +298,43 @@ void AppCommandlineArgs::_buildSplitPaneParser()
     setupSubcommand(_newPaneCommand);
     setupSubcommand(_newPaneShort);
 }
+// Method Description:
+// - Adds the `move-pane-to-tab` subcommand and related options to the commandline parser.
+// - Additionally adds the `mptt` subcommand, which is just a shortened version of `move-pane-to-tab`
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppCommandlineArgs::_buildMovePaneToTabParser()
+{
+    _movePaneToTabCommand = _app.add_subcommand("move-pane-to-tab", RS_A(L"CmdMovePaneToTabDesc"));
+    _movePaneToTabShort = _app.add_subcommand("mptt", RS_A(L"CmdMPTTDesc"));

```

Comment:
One of the other things I was thinking about while implementing this is, especially forward looking, that you could have a target pane, and a `SplitStyle`. E.g. if directional split and drag and drop is added, the underlying command might be like "move pane to tab 2, with target pane 5, split and place on the left side." That is at least a couple of features away, so I don't know how much it makes sense it build that in quite yet. 

As for the command line interface for such a feature, I'm not sure if it needs to be so involved. As a vim user, I like to think of commands and movements. things like `move-focus` and `move-pane` are commands that operate on a movement, whereas "move-pane-to-tab" seems more self contained. Maybe I'm just not creative enough and "move-pane-to-tab previous" or "move-pane-to-tab down" are a suitable interface.

Reply:
Would it be fine to just merge the command line portion and have that call the separate actions, or should there be a higher level of integration?

Reply:
Okay so taking a look at the [man pages](https://man7.org/linux/man-pages/man1/tmux.1.html):

<details>
<summary>tmux man page selection</summary>

```
     join-pane [-bdfhv] [-l size] [-s src-pane] [-t dst-pane]
                   (alias: joinp)
             Like split-window, but instead of splitting dst-pane and
             creating a new pane, split it and move src-pane into the
             space.  This can be used to reverse break-pane.  The -b
             option causes src-pane to be joined to left of or above
             dst-pane.
     move-pane [-bdfhv] [-l size] [-s src-pane] [-t dst-pane]
                   (alias: movep)
             Does the same as join-pane.
     select-pane [-DdeLlMmRUZ] [-T title] [-t target-pane]
                   (alias: selectp)
             Make pane target-pane the active pane in its window.  If
             one of -D, -L, -R, or -U is used, respectively the pane
             below, to the left, to the right, or above the target pane
             is used...
     swap-pane [-dDUZ] [-s src-pane] [-t dst-pane]
                   (alias: swapp)
             Swap two panes.  If -U is used and no source pane is
             specified with -s, dst-pane is swapped with the previous
             pane (before it numerically); -D swaps with the next pane
             (after it numerically).
```
</details>

So they definitely don't allow `move-pane -t 4 left`, because that's nonsensical. "move the pane to tab 4, then move it left"? 

Maybe we should follow their lead on this one. 
* `swap-pane [up|down|left|right|prev|nextInOrder|prevInOrder]|[-t,--target tgt-pane]`
  * This is what we've already got as `move-pane`, so this would be a rename, but without a short form (sad, but okay)
  * obviously, we'd add the `nextInOrder` (etc) values after #10909
  * and we'd add the `-t` argument in another PR, as an exclusive arg with the direction
* `move-pane [-t,--tab index]`: move the pane to the given tab. If `index` is `new`, then moves it to a new tab.
  * in the future we'd add `[-w,--window id]` for moving to another window. 
  * we could add additional params for like, `[-d,--destination pane-id]`: split the given destination pane rather than the active pane in the tab

and I suppose with both we could ass a `[-s,--source pane-id]` arg as well.

I might just be openly brainstorming at this point. What do you think? we haven't shipped `move-pane` yet, so we're definitely still free to change it if we feel there's a better name

Reply:
> Would it be fine to just merge the command line portion and have that call the separate actions, or should there be a higher level of integration?

Yea we definitely thought it was okay to do that. Having two actions internally for one subcommand is _kinda weird_, but is definitely easier to deal with logically than trying to combine them into one args blob with optional properties and all that.

Reply:
I'm happy to rename `move-pane` to `swap-pane` and `move-pane-to-tab` to `move-pane,` or to just make it all `move-pane` just let me know. Bit annoying to dig through all the resource files again but its not too bad.

Reply:
FWIW, the team did feel that we should just do it all on `move-pane`, I've just been brainstorming myself over here since re-reading this thread this morning.

I'm trying to think of all the moving things we want to do with panes, and coming up with a single command that looks like... this?
```
move-pane(mp) [--window,-w window-id] [--source,-s src-pane] [--tab,-t index]|[up|down|left|right|prev|nextInOrder|prevInOrder]|[--destination,-d dest-pane]
```

* we want to move panes to other tabs
* we want to move panes to other windows
* we want to swap a pane with another pane, by ID
* we want to move a pane in a given direction (we want to swap it in a direction)
* we want to be able to move the active pane, or a specific pane in a tab

and I think all that together means that we want to 
1. swap with a (source)? and (a tgt or a direction)
2. move with a (source)? and (a tab)? and (a tgt pane)?

and if these were all on one command, then I don't think it works.  We'd need `tab` and `direction` to be exclusive to differentiate between 1 and 2. But then `direction` and `dest` would need to be exclusive for swap, but not exclusive for move, and now the rules don't work.

So in conclusion, yea:

> rename `move-pane` to `swap-pane` and `move-pane-to-tab` to `move-pane`

(sorry for the runaround on all this!)